src/LogicGate1.java

```diff
@@ -0,0 +1,190 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Map;
+
+/**
+ * {@code LogicGate} represented as a connectable node. Interacts with other gates.
+ * 
+ * 
+ * 
+ * @convention this is empty and not connected to any other gates or this connects to and/or is connected to by other gates and has a full truth table 
+ * @correspondance this is represented as a nonhierarchical connection of gates
+ */
+public class LogicGate1 {
+
+    /*
+     * The logic gates leading into this one.
+     */
+    private ArrayList<LogicGate1> inputs = new ArrayList<>();
```

Comment:
Having four fields of this complexity seems like a lot. This may be fine, but I worry a bit about the complexity.